Source/OxyPlot/Series/AreaSeries.cs

```diff
@@ -118,7 +119,9 @@ protected List<DataPoint> ActualPoints2
         {
             get
             {
-                return this.ItemsSource != null ? this.itemsSourcePoints2 : this.points2;
+                return this.ItemsSource != null 
+                    ? this.itemsSourcePoints2 
+                    : (this.points2.Any() ? this.points2 : this.Points.Select(p => new DataPoint(p.X, ConstantY2)).ToList());
```

Comment:
Only use ToList() for the `Select()` clause as `this.points2` is a list already.
Also I think the points should be cached in `itemsSourcePoints2` similar to the implementation of DataPointSeries.